<?xml version="1.0" ?>
<gazebo version="1.2">
  <world name="default">
    <scene>
      <ambient>0.5 0.5 0.5 1</ambient>
      <background>0.5 0.5 0.5 1</background>
      <shadows>false</shadows>
    </scene>
    <include filename="sun.light" />
    <physics type="ode">
      <gravity>0.0 0.0 -9.81</gravity>
      <ode>
        <solver>
          <type>quick</type>
          <dt>0.001</dt>
          <iters>1000</iters>
          <sor>1.0</sor>
        </solver>
        <constraints>
          <cfm>0</cfm>
          <erp>1.0</erp>
          <contact_max_correcting_vel>0.0</contact_max_correcting_vel>
          <contact_surface_layer>0.0</contact_surface_layer>
        </constraints>
      </ode>
    </physics>
    <model name="model_1">
      <origin>0.0 0.0 10.1 1.57079633 0.0 0.0</origin>
      <link name="link_1">
        <origin>0.0 0.0 0.0 0.0 0.0 0.0</origin>
        <inertial>
          <origin>0.0 0.0 -10.0 0.0 0.0 0.0</origin>
          <inertia>
            <ixx>0.0</ixx>
            <ixy>0.0</ixy>
            <ixz>0.0</ixz>
            <iyy>0.0</iyy>
            <iyz>0.0</iyz>
            <izz>0.0</izz>
          </inertia>
          <mass>5.0</mass>
        </inertial>
        <damping>
          <linear>0.0</linear>
          <angular>0.0</angular>
        </damping>
        <visual name="visual_cylinder">
          <origin>0.0 0.0 -5.0 0.0 0.0 0.0</origin>
          <geometry>
            <cylinder>
              <radius>0.05</radius>
              <length>10.0</length>
            </cylinder>
          </geometry>
          <material>
            <script>Gazebo/Green</script>
          </material>
        </visual>
        <collision name="collision_cylinder">
          <origin>0.0 0.0 -5.0 0.0 0.0 0.0</origin>
          <max_contacts>250</max_contacts>
          <geometry>
            <cylinder>
              <radius>0.05</radius>
              <length>10.0</length>
            </cylinder>
          </geometry>
        </collision>
        <gravity>true</gravity>
        <self_collide>true</self_collide>
        <kinematic>false</kinematic>
      </link>
      <joint name="joint_0" type="revolute">
        <parent>world</parent>
        <child>link_1</child>
        <origin>0.0 0.0 0.0 0.0 0.0 0.0</origin>
        <axis>
          <xyz>1.0 0.0 0.0</xyz>
        </axis>
      </joint>
      <link name="link_2">
        <origin>0.0 0.0 -10.0 0.0 0.0 0.0</origin>
        <inertial>
          <origin>0.0 0.0 0.0 0.0 0.0 0.0</origin>
          <inertia>
            <ixx>0.0</ixx>
            <ixy>0.0</ixy>
            <ixz>0.0</ixz>
            <iyy>0.0</iyy>
            <iyz>0.0</iyz>
            <izz>0.0</izz>
          </inertia>
          <mass>5.0</mass>
        </inertial>
        <damping>
          <linear>0.0</linear>
          <angular>0.0</angular>
        </damping>
        <visual name="visual_sphere">
          <origin>0.0 0.0 0.0 0.0 0.0 0.0</origin>
          <geometry>
            <sphere>
              <radius>0.1</radius>
            </sphere>
          </geometry>
          <material>
            <script>Gazebo/Red</script>
          </material>
        </visual>
        <collision name="collision_sphere">
          <origin>0.0 0.0 0.0 0.0 0.0 0.0</origin>
          <max_contacts>250</max_contacts>
          <geometry>
            <sphere>
              <radius>0.1</radius>
            </sphere>
          </geometry>
        </collision>
        <gravity>true</gravity>
        <self_collide>true</self_collide>
        <kinematic>false</kinematic>
      </link>
      <joint name="joint_1" type="revolute">
        <parent>link_1</parent>
        <child>link_2</child>
        <origin>0.0 0.0 0.0 0.0 0.0 0.0</origin>
        <axis>
          <limit lower="0" upper="0" effort="0" velocity="0" />
          <xyz>0 0 1</xyz>
        </axis>
      </joint>
      <static>false</static>
    </model>
    <model name="model_2">
      <origin>1.0 0.0 10.1 1.57079633 0.0 0.0</origin>
      <link name="link_1">
        <origin>0.0 0.0 0.0 0.0 0.0 0.0</origin>
        <inertial>
          <origin>0.0 0.0 -10.0 0.0 0.0 0.0</origin>
          <inertia>
            <ixx>0.0</ixx>
            <ixy>0.0</ixy>
            <ixz>0.0</ixz>
            <iyy>0.0</iyy>
            <iyz>0.0</iyz>
            <izz>0.0</izz>
          </inertia>
          <mass>10.0</mass>
        </inertial>
        <damping>
          <linear>0.0</linear>
          <angular>0.0</angular>
        </damping>
        <visual name="visual_cylinder">
          <origin>0.0 0.0 -5.0 0.0 0.0 0.0</origin>
          <geometry>
            <cylinder>
              <radius>0.1</radius>
              <length>10.0</length>
            </cylinder>
          </geometry>
          <material>
            <script>Gazebo/Green</script>
          </material>
        </visual>
        <collision name="collision_cylinder">
          <origin>0.0 0.0 -5.0 0.0 0.0 0.0</origin>
          <max_contacts>250</max_contacts>
          <geometry>
            <cylinder>
              <radius>0.1</radius>
              <length>10.0</length>
            </cylinder>
          </geometry>
        </collision>
        <gravity>true</gravity>
        <self_collide>true</self_collide>
        <kinematic>false</kinematic>
      </link>
      <joint name="joint_0" type="revolute">
        <parent>world</parent>
        <child>link_1</child>
        <origin>0.0 0.0 0.0 0.0 0.0 0.0</origin>
        <axis>
          <xyz>1.0 0.0 0.0</xyz>
        </axis>
      </joint>
      <static>false</static>
    </model>
    <model name="model_3">
      <origin>0.0 -10.5 10.1 0.0 1.57079633 0.0</origin>
      <link name="link_1">
        <origin>0.0 0.0 0.0 0.0 0.0 0.0</origin>
        <inertial>
          <origin>0.0 0.0 -10.0 0.0 0.0 0.0</origin>
          <inertia>
            <ixx>0.0</ixx>
            <ixy>0.0</ixy>
            <ixz>0.0</ixz>
            <iyy>0.0</iyy>
            <iyz>0.0</iyz>
            <izz>0.0</izz>
          </inertia>
          <mass>10.0</mass>
        </inertial>
        <damping>
          <linear>0.0</linear>
          <angular>0.0</angular>
        </damping>
        <visual name="visual_cylinder">
          <origin>0.0 0.0 -5.0 0.0 0.0 0.0</origin>
          <geometry>
            <cylinder>
              <radius>0.1</radius>
              <length>10.0</length>
            </cylinder>
          </geometry>
          <material>
            <script>Gazebo/Green</script>
          </material>
        </visual>
        <collision name="collision_cylinder">
          <origin>0.0 0.0 -5.0 0.0 0.0 0.0</origin>
          <max_contacts>250</max_contacts>
          <geometry>
            <cylinder>
              <radius>0.1</radius>
              <length>10.0</length>
            </cylinder>
          </geometry>
        </collision>
        <gravity>true</gravity>
        <self_collide>true</self_collide>
        <kinematic>false</kinematic>
      </link>
      <joint name="joint_0" type="revolute">
        <parent>world</parent>
        <child>link_1</child>
        <origin>0.0 0.0 0.0 0.0 0.0 0.0</origin>
        <axis>
          <xyz>0.0 1.0 0.0</xyz>
        </axis>
      </joint>
      <static>false</static>
    </model>
  </world>
</gazebo>
